csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -100,6 +101,38 @@ public static SessionOptions MakeSessionOptionWithTensorrtProvider(int deviceId
             }
         }
 
+        /// <summary>
+        /// A helper method to construct a SessionOptions object for TensorRT execution provider.
+        /// Use only if CUDA/TensorRT are installed and you have the onnxruntime package specific to this Execution Provider.
+        /// </summary>
+        /// <param name="trtProviderOptions">TensorRT EP provider options</param>
+        /// <returns>A SessionsOptions() object configured for execution on provider options</returns>
+        public static SessionOptions MakeSessionOptionWithTensorrtProvider(OrtTensorRTProviderOptions trtProviderOptions)
+        {
+            CheckTensorrtExecutionProviderDLLs();
+            SessionOptions options = new SessionOptions();
+            try
+            {
+                // get device id for configuring CUDA EP
+                int deviceId;
+                string optionsStr;
+                var dict = new Dictionary<string, string>();
+                optionsStr = trtProviderOptions.GetOptions();
+                ProviderOptionsValueHelper.StringToDict(optionsStr, dict);
+                deviceId = Int32.Parse(dict["device_id"]);
```

Comment:
The usage seems to be a little convoluted.
1) construct a dictionary with all options
2) call UpdateOptions
3) construct a dictionary to get options back to as string form
4) convert to int and then call the API
Would it be better to make sure that we already have an int ready in the UpdateOptions and store it bc we know we will need it?

Reply:
fixed.